sqlfs/writer.go

```diff
@@ -56,31 +51,26 @@ func (w *Writer) Write(p []byte) (n int, e error) {
 // Close the connection of the sqlfs
 func (w *Writer) Close() error {
 	if e := w.flush(); e != nil {
-		return fmt.Errorf("writer flush failed: %v", e)
-	}
-	if w.insert != nil {
-		if e := w.insert.Close(); e != nil {
-			return e
-		}
-		w.insert = nil // Mark closed.
+		return fmt.Errorf("close failed: %v", e)
 	}
-	w.db = nil // Mark closed.
+	w.db = nil // mark closed
 	return nil
 }
 
 func (w *Writer) flush() error {
-	var e error
-	if w.insert == nil {
-		w.insert, e = w.db.Prepare(
-			fmt.Sprintf("INSERT INTO %s (block) VALUES(?)", w.table))
-		if e != nil {
-			return fmt.Errorf("flush failed to prepare insert %s: %v", w.table, e)
-		}
+	if w.db == nil {
+		return fmt.Errorf("bad database connection")
 	}
-	_, e = w.insert.Exec(w.buf)
-	if e != nil {
-		return fmt.Errorf("flush failed to execute insert: %v", e)
+
+	if len(w.buf) > 0 {
+		block := base64.StdEncoding.EncodeToString(w.buf)
+		query := fmt.Sprintf("INSERT INTO %s (id, block) VALUES(%d, '%s')",
+			w.table, w.flushID, block)
+		if _, e := w.db.Exec(query); e != nil {
+			return fmt.Errorf("flush to %s, error:%v", w.table, e)
+		}
+		w.buf = w.buf[:0]
+		w.flushID++
```

Comment:
Just curious that if we are writing multiple versions of model parameters to DB ( have many flushIDs) then how to determine which model to use when running prediction?

Reply:
Is there already checks that can enure allways writing parameters of the same model? Like if a user trained a `DNN` model first, then he change the model type to `LR` but didn't change the table name where we save the model parameters?

Reply:
We are considering using [independent storage media](https://github.com/sql-machine-learning/sqlflow/issues/377) to save models. Let's keep on discussing in that thread.